lib/httpClient.js

```diff
@@ -112,10 +112,11 @@ function Client (opts) {
 
     this.emit('response', resp.headers.statusCode, resp.bytes, resp.duration, this.rate)
 
+    const rpi = (this.cer + pipelining) % pipelining
```

Comment:
rpi = request pipelining index
cer = current expected response

Separated `rpi` and `cer` because all pipelining requests are fired at once without waiting for response and the following happens (with pipelining = 2)

- Both requests fired
- `cer` is **0**
- Response for first came
- After processing it, we call `_doRequest` with incremented `cer`
- Incremented `cer` is now **1** 
- `_doRequest` tries to fire the same request again, hence re-setting the startTime to now which messes up the `duration` calculation when response comes.

Separating `rip` ensures that after getting response for **first** request, we fire **first** one again because second is already running right now.

Does it make sense? 😅

Reply:
Totally, put it in a comment ;).

Reply:
All of this code was written at a time when I understood function character length had a direct impact on it being optimised. Newer versions of v8 are not limited by this afaik. If I was refactoring this code now I would recommend rewriting the variables as their full names, and adding many, many more comments.

`rpi` = `requestPipeliningIndex`
`cer` = `currentExpectedResponse`

After looking back at this code for the first time in years, I was struggling to understand the variable meanings myself. I would even being to rename many functions (even anonymous ones), to give the code more contextual meaning. I'd aim to make it readable/understandable first, then measure the impact. My biggest mistake writing the code in autocannon was not doing that during initial implementations, so I'm advising to learn from that 😅 

I am happy to refactor this in a follow up PR, if you'd like.

Reply:
@salmanm your explanation above makes sense to me, I just want to make sure there isn't an implicit assumption in this statement:

> Separating rip ensures that after getting response for first request, we fire first one again because second is already running right now.

This is not assuming that requests will take similar times to execute, right? In other words, this supports the case whereby, with pipelining 2:

- fire first request
- fire second request
- **second  response comes back** before the first
- first response comes back